README.md

```diff
@@ -57,89 +53,223 @@ The following code creates a KNN index with fields my_vector1, my_vector2, my_ve
 ```
 PUT /myindex
 {
-  "settings" : {
-    "index": {
-      "knn": true
+    "settings" : {
+        "number_of_shards" :   1,
+        "number_of_replicas" : 0,
+        "index": {
+            "knn": true
+        }
+    },
+    "mappings": {
+        "properties": {
+            "my_vector": {
+                "type": "knn_vector",
+                "dimension": 2
+            }
+        }
     }
-  },
-  "mappings": {
-    "my_images": {
-      "properties": {
-        "my_vector1": {
-          "type": "knn_vector"
-        },
-        "my_vector2": {
-          "type": "knn_vector"
-        },
-        "my_vector3": {
-        "type": "knn_vector"
-      }
-    }
-  }
 }
 ```
 
 * Indexing sample docs to KNN index
 
 ```
-curl -X POST "localhost:9200/myindex/_doc/1" -H 'Content-Type: application/json' -d'
+PUT /myindex/_doc/2?refresh=true
 {
-"my_vector" : [1.5, 2.5],
-"price":10
+    "my_vector" : [1.5, 2.5],
+    "price":10
 }
-'
+```
+
+* Querying K-Nearest neighbors
 
-curl -X PUT "localhost:9200/myindex/_doc/2" -H 'Content-Type: application/json' -d'
+```
+POST /myindex/_search
 {
-"my_vector" : [2.5, 3.5],
-"price":12
+    "size" : 10,
+    "query": {
+        "knn": {
+            "my_vector": {
+                "vector": [3, 4],
+                "k": 2
+            }
+        }
+    }
 }
-'
+```
+
+## Java Native library usage
+For plugin installations from archive(.zip), it is necessary to ensure ```.so``` file for linux OS and ```.jnilib``` file for Mac OS are present in the java library path. This can be possible by copying .so/.jnilib to either $ES_HOME or by adding manually ```-Djava.library.path=<path_to_lib_files>``` in ```jvm.options``` file
+
+## Settings
+### Index Level Settings
+KNN Index level settings should be provided at the time of the index creation. If the settings are not provided, default values will be use. These settings are static which means they cannot be changed after index creation.
```

Comment:
You must provide index-level settings when you create the index. If you don't provide these settings, KNN uses its default values. These settings are static, which means you can't modify them after index creation.

Reply:
Will update thanks